.github/workflows/test.yml

```diff
@@ -0,0 +1,79 @@
+name: Test & lint
+on: [push]
+
+env:
+  RAILS_ENV: test
+  PGHOST: localhost
+  PGUSER: postgres
+
+jobs:
+  tests:
+    name: Test & lint
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ['2.4', '2.5', '2.6', '2.7', 'jruby']
+        gemfile: ['gemfiles/activerecord_4.2.0.gemfile', 'gemfiles/activerecord_5.0.2.gemfile', 'gemfiles/activerecord_5.1.0.gemfile', 'gemfiles/activerecord_5.2.2.gemfile', 'gemfiles/activerecord_6.0.0.gemfile', 'gemfiles/activerecord_6.1.0.gemfile', 'gemfiles/activerecord_master.gemfile']
+        exclude:
+          - gemfile: 'gemfiles/activerecord_4.2.0.gemfile'
+            ruby: '2.7' # rails 4.2 can't run on ruby 2.7 due to BigDecimal API change
+          - gemfile: 'gemfiles/activerecord_6.0.0.gemfile'
+            ruby: '2.4' # rails 6+ requires ruby 2.5+
+          - gemfile: 'gemfiles/activerecord_6.1.0.gemfile'
+            ruby: '2.4' # rails 6+ requires ruby 2.5+
+          - gemfile: 'gemfiles/activerecord_master.gemfile'
+            ruby: '2.4' # rails 6+ requires ruby 2.5+
+          - gemfile: 'gemfiles/activerecord_5.0.2.gemfile'
+            ruby: 'jruby' # this *should* work - there's a test failure; it's not incompatible like the other excludes. could be an issue in Rails 5.0.2?
+          - gemfile: 'gemfiles/activerecord_6.1.0.gemfile'
+            ruby: 'jruby' # this *should* work. it seems like there's an issue with rails 6 on jruby.
+          - gemfile: 'gemfiles/activerecord_master.gemfile'
+            ruby: 'jruby' # this *should* work. it seems like there's an issue with rails 6 on jruby.
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_DB: cancan_postgresql_spec
+          POSTGRES_PASSWORD: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports: ["5432:5432"]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: '20'
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+
+      - name: Install bundler
+        run: gem install bundler
+
+      # https://github.com/oracle/truffleruby/issues/62#issuecomment-362065726
+      # TODO: can't get this to work on Github actions. see comments in https://github.com/CanCanCommunity/cancancan/pull/669
+      - name: Nokogiri support for Truffleruby
+        run: BUNDLE_GEMFILE=${{ matrix.gemfile }} bundle config build.nokogiri --use-system-libraries
+        if: ${{ matrix.ruby == 'truffleruby' }}
+
+      - name: Install gems
+        run: BUNDLE_GEMFILE=${{ matrix.gemfile }} bundle install --jobs 2
```

Comment:
It's better to set `BUNDLE_GEMFILE` in the `env:` of the job, as described in https://github.com/ruby/setup-ruby#caching-bundle-install-automatically